pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -308,3 +326,63 @@ func createValidImageVersionLabel(image string) string {
 	}
 	return cephImageVersion
 }
+
+func waitForPvcToExpandWithTimeout(ctx context.Context, client client.Client, pvcDesiredAndActualSizeMap map[types.NamespacedName]pvcSize, pvcResizeWaitTimeOut time.Duration) {
+	currentTime := time.Now()
+	if pvcResizeWaitTimeOut == 0 {
+		pvcResizeWaitTimeOut = defaultWaitTimeoutForHealthyOSD
+	}
+	timeout := currentTime.Add(pvcResizeWaitTimeOut)
+	allPvcResize := false
+	for !allPvcResize {
+		if time.Now().After(timeout) {
+			logger.Errorf("timed out waiting for PVC size to be updated to %v", pvcDesiredAndActualSizeMap)
+			return
+		}
+		// wait for 5 seconds before checking again
+		time.Sleep(5 * time.Second)
+
+		allPvcResize = checkAllPvcResize(ctx, client, pvcDesiredAndActualSizeMap)
+	}
+}
+
+func checkAllPvcResize(ctx context.Context, client client.Client, pvcDesiredAndActualSizeMap map[types.NamespacedName]pvcSize) bool {
+	numberOfPvcResize := 0
+
+	for pvcNamespaceName, pvc := range pvcDesiredAndActualSizeMap {
+		desiredSize := pvc.desiredSize
+		actualSize := pvc.actualSize
+		if desiredSize.Value() == actualSize.Value() {
+			numberOfPvcResize++
+			continue
+		}
+
+		// get the latest PVC
+		currentPVC := &v1.PersistentVolumeClaim{}
+		err := client.Get(ctx, types.NamespacedName{Name: pvcNamespaceName.Name, Namespace: pvcNamespaceName.Namespace}, currentPVC)
+		if err != nil {
+			logger.Errorf("failed to get PVC %q. %v", currentPVC.Name, err)
+			continue
+		}
+
+		// check if the size is updated
+		pvcSize := currentPVC.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(desiredSize) != 0 {
+			logger.Infof("PVC %q size is not updated to %s %s", currentPVC.Name, pvcSize.String(), desiredSize.String())
+			continue
+		}
+
+		if currentPVC.Status.Capacity != nil {
+			if currentPVC.Status.Capacity[v1.ResourceStorage] == pvcSize {
+				logger.Infof("PVC %q size is  updated to %s", currentPVC.Name, pvcSize.String())
+				numberOfPvcResize++
```

Comment:
Need a continue here? Otherwise, does the logging on line 384 make sense?
```suggestion
				numberOfPvcResize++
				continue
```